src/fsharp/FSharp.Core/set.fs

```diff
@@ -285,6 +285,9 @@ namespace Microsoft.FSharp.Collections
         let filter comparer f s = filterAux comparer f s SetEmpty
 
         let rec diffAux comparer m acc = 
+            match acc with
+            | SetEmpty -> acc
```

Comment:
We can. But, I don't think it would be a good idea. Because we don't know anything about `m` at this point. So, we should be starting another match for `m`. If you look at the flow, this is actually what it is doing. Morever, if `acc` is indeed `SetOne` it handled in the very first call to `remove`. So, I would suggest not to handle `SetOne` case for `acc` in the first match expression.